chart/tests/test_security_context.py

```diff
@@ -0,0 +1,234 @@
+            assert "git-sync-init" in [c["name"] for c in jmespath.search("spec.template.spec.initContainers", docs[index])]
+            assert 3000 == jmespath.search("spec.template.spec.initContainers[?name=='git-sync-init'].securityContext.runAsUser | [0]", docs[index])
+            assert 3000 == jmespath.search("spec.template.spec.containers[?name=='git-sync'].securityContext.runAsUser | [0]", docs[index])
+
+class TestSecurityContext:
+    # Test podSecurity setting for Pods and Containers
+    def test_check_default_setting(self):
+        docs = render_chart(
+            values={
+                "podSecurity": {
+                    "securityContext": {"runAsUser": 6000, "fsGroup": 60}, 
+                    "containerSecurityContext": {"runAsUser": 4000, "runAsGroup": 40}
+                },
+                "webserver": {"defaultUser": {"enabled": True}},
+                "flower": {"enabled": True},
+                "statsd": {"enabled": True},
+                "airflowVersion": "2.2.0",
+                "executor": "CeleryKubernetesExecutor",
+            },
+            show_only=[
+                "templates/flower/flower-deployment.yaml",
+                "templates/scheduler/scheduler-deployment.yaml",
+                "templates/triggerer/triggerer-deployment.yaml",
+                "templates/webserver/webserver-deployment.yaml",
+                "templates/workers/worker-deployment.yaml",
+                "templates/jobs/create-user-job.yaml",
+                "templates/jobs/migrate-database-job.yaml",
+                "templates/statsd/statsd-deployment.yaml"
+            ],
+        )
+
+        for index in range(len(docs)):
+            print (docs[index])
+            assert 6000 == jmespath.search("spec.template.spec.securityContext.runAsUser", docs[index])
+            assert 60 == jmespath.search("spec.template.spec.securityContext.fsGroup", docs[index])
+            assert 4000 == jmespath.search("spec.template.spec.containers[0].securityContext.runAsUser", docs[index])
+            assert 40 == jmespath.search("spec.template.spec.containers[0].securityContext.runAsGroup", docs[index])
+
+    # Test priority:
+    # <local>.containerSecurityContext > podSecurity.containerSecurityContext > uid + gid
+    # <local>.containerSecurityContext > podSecurity.containerSecurityContext > uid + gid
+    def test_check_local_setting(self):
+        docs = render_chart(
+            values={
+                "uid": 3000,
+                "gid": 30,
+                "podSecurity": {
+                    "securityContext": {"runAsUser": 6000, "fsGroup": 60},
+                    "containerSecurityContext": {"runAsUser": 4000, "runAsGroup": 40}
+                },
+                "webserver": {
+                    "defaultUser": {"enabled": True},
+                    "securityContext": {"runAsUser": 9000, "fsGroup": 90},
+                    "containerSecurityContext": {"runAsUser": 8000, "runAsGroup": 80}
```

Comment:
So we don't have to define these for all of them, maybe pull these out and do something like this instead:

```
# define this above
component_contexts = {
    "securityContext": {"runAsUser": 9000, "fsGroup": 90},
    "containerSecurityContext": {"runAsUser": 8000, "runAsGroup": 80}
}
```

Then change this section (and the rest to):
```suggestion
                    **component_contexts
```